net_utils.go

```diff
@@ -207,6 +207,41 @@ func incrementIP(ip net.IP) (result net.IP) {
 	return
 }
 
+// decrementIP returns the given IP 1 1
+func decrementIP(ip net.IP) (result net.IP) {
+	// copy the ip into the result
+	result = make([]byte, len(ip))
+	copy([]byte(result), []byte(ip))
+
+	// Can't decrement an nil or invalid ip
+	if len(result) != 4 && len(result) != 16 {
+		panic(fmt.Errorf("Bad number of bytes for address: %d", len(result)))
+	}
+	// do subtract
```

Comment:
I don't understand why this has to be split into two sections. Why not make this implementation look like incrementIP? Are you uncomfortable with the use of overflow in that implementation?

```
// decrementIP returns the given IP - 1
func decrementIP(ip net.IP) (result net.IP) {
	result = net.ParseIP("::")
	if len(ip) == 4 {
		result = net.ParseIP("0.0.0.0").To4()
	}

	borrow := true
	for i := len(ip) - 1; i >= 0; i-- {
		result[i] = ip[i]
		if borrow {
			result[i]--
			if result[i] != 255 {
				borrow = false
			}
		}
	}
	return
}
```

Reply:
weird.. it was giving me unhappy messages earlier about overflow. they are gone now.. idk